### ISSUE TITLE
Segmentation violation on stats

### ISSUE THREAD

**Issue by Wouter0100** [restic/restic#2320]
Output of `restic version`
--------------------------
```
# restic version
restic 0.9.5 compiled with go1.12.4 on linux/amd64
```

How did you run restic exactly?
-------------------------------
With the linux amd64 binary moved into a $PATH location and on the backup server. This backup server has of a lot of RAM (although, mostly in it's buffer/cache):
```
# free -h
              total        used        free      shared  buff/cache   available
Mem:            62G        575M        672M        1,7G         61G         59G
Swap:           31G          0B         31G
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Local, in this case from the backup server itself. The repository is fed externally (over SFTP). 

Expected behavior
-----------------
Show the stats of the repository.

Actual behavior
---------------
```
# restic -r repository stats
repository 774c5754 opened successfully, password is correct
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7b82c8]

goroutine 9 [running]:
github.com/restic/restic/internal/repository.(*Index).ID(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/restic/internal/repository/index.go:377 +0x38
github.com/restic/restic/internal/repository.(*Repository).LoadIndex.func5(0x8, 0xe6c288)
	/restic/internal/repository/repository.go:467 +0xce
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0003b2210, 0xc00000c8c0)
	/restic/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x57
created by golang.org/x/sync/errgroup.(*Group).Go
	/restic/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66
```

Steps to reproduce the behavior
-------------------------------
Run the following command:
```
# restic -r repository stats
```

I haven't found a way to reproduce this error in one of our other repositories. This is by far the largest one, with ~3615 snapshots right now and growing (probably to ~10000 snapshots).

Do you have any idea what may have caused this?
-----------------------------------------------
I guess the large amount of snapshots, but I'm not entirely sure.

Do you have an idea how to solve the issue?
-------------------------------------------
Unfortunately not, currently.

Did restic help you or made you happy in any way?
-------------------------------------------------
Yes, Restic has been awesome so far in providing a robuust backup solution! We deployed it on multiple environments and are using it with pleasure! Restic is great :)

**Comment by todd1794** [comment]
Thank you for reporting this same thing happened to me. I got a around it with adding a --no-cache.

**Comment by fin-ger** [comment]
`rm -rf .cache/restic` solved the problem for me. Remember to delete the cache for all users using `restic` (e.g. root).

**Comment by fd0** [comment]
This error is fixed in #2321 which is included in restic 0.9.6, I'll therefore close this issue for now. Please report a new issue (including the stack trace!) if it happens again with 0.9.6 or newer. Thanks a lot!
